src/Calypso-SystemTools-FullBrowser/ClyFullBrowser.class.st

```diff
@@ -789,13 +790,19 @@ ClyFullBrowser >> switchClassViewTo: aQueryResult inScope: aScope [
 
 { #category : #navigation }
 ClyFullBrowser >> switchToFlatClasses [
-	
-	| classQueries |
+	| classQueries fullQuery dataSource |
 	classQueries := OrderedCollection new.
-	packageView selection groupItemsByTypeAndDo: [ :typeClass :items | 
-		classQueries addAll: (typeClass prepareClassQueriesFrom: items in: navigationEnvironment)].
-	
-	classView showQueries: classQueries as: ClyExtensionLastSortedClasses hierarchical
```

Comment:
I think this code should be kept. But the widget (classView) needs to be configured what type of data source to use in the classView initialization:
```Smalltalk
classView := self newNavigationView
    ...
    enableCollapsingOfPreexpandedItems.
```
This way you will fix hierarchy mode as well.

Reply:
Okay, I debated about adding the capability to hierarchy mode, but it seems like it's not a very wide tree - mostly (or all?) one child per parent, no? Doesn't matter to me. If you think I should add it I will. And in case you do, just to clarify, you're suggesting that I simply add that one `#enableCollapsingOfPreexpandedItems` send, correct?

Reply:
> Okay, I debated about adding the capability to hierarchy mode, but it seems like it's not a very wide tree - mostly (or all?) one child per parent, no? Doesn't matter to me. If you think I should add it I will. 

I think for consistency classes should be always collapsable (or always not) independently of mode. 

> And in case you do, just to clarify, you're suggesting that I simply add that one `#enableCollapsingOfPreexpandedItems` send, correct

I mean that this configuration needs to be implemented in ClyQueryViewMorph. Then we will not need to touch a query API of this widget. We will just enable this behaviour in classView initialization.

Reply:
Sounds good, but I have no idea how to translate the fix to what you suggested!

Reply:
@dionisiydk Done. See second commit.